js/package.json

```diff
@@ -26,14 +47,15 @@
     "//": "Requires tinygo 0.38.0 or later",
     "build": "npm run build-go && npm run build-js",
     "build-go": "tinygo build -o format.wasm -target wasm --no-debug",
-    "build-js": "tsc && cp format.wasm wasm_exec.js dist",
-    "format": "prettier --write \"**/*.{js,ts,json}\"",
+    "build-js": "tsc && r -f cjs -o dist -x '**/wasm_exec.js' && cp index.d.cts format.wasm wasm_exec.js dist",
+    "format": "prettier --write \"**/*.{cts,js,ts,json}\"",
     "test": "echo \"Error: no test specified\" && exit 1"
   },
   "devDependencies": {
+    "@pkgr/rollup": "^6.0.2",
```

Comment:
Why do we need rollup?

Reply:
This PR bundle cjs?

Reply:
I see. Maybe a separate build would work:
https://tduyng.com/blog/dual-package-typescript/

```
npx tsc -p tsconfig.lib.json --module NodeNext --outDir lib/esm
echo '{"type": "module"}' > lib/esm/package.json

npx tsc -p tsconfig.lib.json --module CommonJS --moduleResolution Node --outDir lib/cjs
echo '{"type": "commonjs"}' > lib/cjs/package.json

echo 'Compilation successful'
```

I'll see if I can play around with this later.